README.md

```diff
@@ -1,9 +1,98 @@
 # edgex-checkbox-provider
 
-This project contains the Checkbox tests of the [EdgeX Foundry](https://docs.edgexfoundry.org/) snaps.
+## Introduction
 
-The upstream repository is hosted on Github:  
-https://github.com/canonical/edgex-checkbox-provider
+This project contains the checkbox tests of the [edgex foundry](https://docs.edgexfoundry.org/) snaps. 
+[Checkbox](https://checkbox.readthedocs.io/en/latest/) is a test automation software performed by the Canonical certification team. 
+The [upstream repository](/../../) is hosted on Github; 
+The tests run via the [Launchpad mirror](https://code.launchpad.net/checkbox-provider-edgex).
 
-The tests run via the Launchpad mirror at:  
-https://code.launchpad.net/checkbox-provider-edgex
+When edgexfoundry snap is released to the latest/beta channel, the corresponding checkbox test will be triggered. 
+
+## Usage
+### Run tests using checkbox-edgexfoundry snap in isolation (recommend)
+```bash
+sudo snap install checkbox-edgexfoundry --devmode --channel=latest/edge
+```
+checkbox-edgexfoundry snap depends on extra environment variables, so setting test channel and release name:
+```bash
+sudo DEFAULT_TEST_CHANNEL=<"channel"> checkbox-edgexfoundry.<release name>
+# Here is an example
+sudo DEFAULT_TEST_CHANNEL="2.1/beta" checkbox-edgexfoundry.jakarta
+
+```
+### Run tests using checkbox CLI
+```bash
+cd edgex-checkbox-provider/
+sudo ./manage.py install
+checkbox-cli
+```
+![image-checkbox-test-plans](./assets/checkbox-test-plans.png)
+
+Press SPACE to select an EdgeX test plan, then choose tests to run on your system:
+
+![image-edgex-test-plan](./assets/edgex-test-plan.png)
+
+### Run tests locally
+Remove these snaps to avoid missing edgex-secretstore-token issue, if applicable:
+```bash
+snap remove --purge <edgex-device-service>
+snap remove --purge <edgex-app-service>
+```
+Get the edgex-checkbox-provider:
+```bash
+cd edgex-checkbox-provider/data/jakarta
+```
+Download a specific version snap:
+```bash
+snap download edgexfoundry <channel>
+sudo ./run-all-tests-locally.sh -s <edgexfoundry.snap> -t <single test>
+```
+Here are some examples:
+
+```bash
+# run a single test using locally built snaps
+snap download edgexfoundry --channel=2.1/stable
+sudo ./run-all-tests-locally.sh -s edgexfoundry_3375.snap -t test-rules-engine.sh
+```
+```bash
+# run all tests using locally built snaps
+sudo ./run-all-tests-locally.sh-s edgexfoundry_3375.snap
+```
+```bash
+# run a single test using snap from latest/beta channel
+sudo ./run-all-tests-locally.sh -t test-rules-engine.sh
+```
+```bash
+# run all tests using snap from latest/beta channel
+sudo ./run-all-tests-locally.sh
+```
+```bash
+# run tests showing output of tests even if passed
+sudo ./run-all-tests-locally.sh -t -v
+sudo ./run-all-tests-locally.sh -t test-install.sh -v
+```
+```bash
+# get a list of available options
+sudo ./run-all-tests-locally.sh -h
+```
```

Comment:
The last three lines are really enough, followed by one or two examples:

Get a list of available options:
```
sudo ./run-all-tests-locally.sh -h
```

For example, to run a single test with a local snap:
```bash
sudo ./run-all-tests-locally.sh -s edgexfoundry.snap -t test-rules-engine.sh
```